src/lib/features/dependent-features/dependent-features-controller.ts

```diff
@@ -49,13 +59,15 @@ export default class DependentFeaturesController extends Controller {
         config: IUnleashConfig,
         {
             transactionalDependentFeaturesService,
+            dependentFeaturesService,
             openApiService,
         }: DependentFeaturesServices,
         startTransaction: TransactionCreator<UnleashTransaction>,
     ) {
         super(config);
         this.transactionalDependentFeaturesService =
             transactionalDependentFeaturesService;
+        this.dependentFeaturesService = dependentFeaturesService;
```

Comment:
deletion does not have to be transactional as there's only one DB operation involved